lua/obsidian/workspace.lua

```diff
@@ -0,0 +1,71 @@
+---@class obsidian.Workspace
+---@field name string
+---@field path string
+---@return obsidian.Workspace
+local workspace = {}
+
+---Create a new workspace
+---
+---@param name string
+---@param path string
+---
+---@return obsidian.Workspace
+workspace.new = function(name, path)
+  local self = setmetatable({}, { __index = workspace })
+
+  self.name = name
+  self.path = vim.fs.normalize(path)
+
+  return self
+end
+
+workspace.new_from_cwd = function()
+  return workspace.new(".", vim.fn.getcwd())
+end
+
+---Determines if cwd is a workspace
+---
+---@param workspaces table<obsidian.Workspace>
+---@return obsidian.Workspace|nil
+workspace.get_workspace_from_cwd = function(workspaces)
+  local cwd = vim.fn.getcwd()
+  local _, value = next(vim.tbl_filter(function(w)
+    if w.path == cwd then
```

Comment:
We might need to normalize these paths before comparing them, otherwise it seems a bit brittle.

Reply:
See [my comment](https://github.com/epwalsh/obsidian.nvim/pull/155#discussion_r1361333890) in a previous suggestion